tuples_test.go

```diff
@@ -294,6 +295,22 @@ func TestParseEntity(t *testing.T) {
 			ID:       "some.user-name+suffix@some.domain-name+suffix",
 			Relation: "member",
 		},
+	}, {
+		about:        "wildcard entity is parsed correctly",
+		entityString: "user:*",
+		expectedEntity: ofga.Entity{
+			Kind:     "user",
+			ID:       "*",
+			Relation: "",
+		},
+	}, {
+		about:        "wildcard entity with relation is parsed correctly",
+		entityString: "user:*#member",
+		expectedEntity: ofga.Entity{
+			Kind:     "user",
+			ID:       "*",
+			Relation: "member",
+		},
```

Comment:
Feel free to ignore it, but I'd add another test case that verifies parsing something like `user:*blahblah` returns an error.

Reply:
So unfortunately, with our current regex expression, `user:*foo` gets parsed into an entity with 
```
Kind: "user",
ID: "*",
Relation: "",
```
I tried to explain this in the PR description. I was able to improve the regex to catch cases like these but that required the use of a negative lookahead which is not supported in Go's stdlib regex parses.

Reply:
I tested your regex on regex101.com and it wouldn't parse such cases. So, it's a bit strange.

Reply:
Which one?

Reply:
I just added an end-of-line anchor at the end of the pattern and now it works as expected:

```
([A-Za-z0-9_][A-Za-z0-9_-]*):([A-Za-z0-9_][A-Za-z0-9_@.+-]*|[*])(#([A-Za-z0-9_][A-Za-z0-9_-]*))?$
```

Reply:
Ohh smart! Thanks for this, I'll add it now

Reply:
Alternatively, 
```
([A-Za-z0-9_][A-Za-z0-9_-]*)((:[*])|(:([A-Za-z0-9_][A-Za-z0-9_@.+-]*)(#([A-Za-z0-9_][A-Za-z0-9_-]*))?))
```
seems to work as well :)

Reply:
That one has the same problem as what I originally had. A string like `user:*foo` doesn't get matched properly.
![image](https://github.com/canonical/ofga/assets/46668016/de19db60-4526-4642-b4fc-44bd5e4cda3d)

Reply:
You are right, my bad!